### PR TITLE
Fix repository listing page to only show results for a single repo.

### LINKED ISSUE #207: Repository listing page shows results for all repositories.
When viewing the listing page on [Heroku](http://lore-ci.herokuapp.com/repositories/animating-stuffed-animals/?q=), I see the courses uploaded to the "Animating Stuffed Animals" repository.

When I view the [milo test](http://lore-ci.herokuapp.com/repositories/milo-repo/) repo, I see the same courses.

The `no_query_found` method of `search.forms.SearchForm` needs to be updated to filter by the repository being viewed.

### PR DESCRIPTION
**Pull request by ShawnMilo**
Previously it was showing all.

fixes #207

### CODE REVIEW

File: search/tests/test_search_view.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        resp = self.client.get(reverse("repositories", args=(self.repo.slug,)))
+        body = resp.content.decode('utf-8')
+        self.assertTrue(self.resource.title in body)
+
+        # Make a new repo with no content.
+        new_repo = create_repo("other", "whatever", self.user.id)
+
+        resp = self.client.get(reverse("repositories", args=(new_repo.slug,)))
+        body = resp.content.decode('utf-8')
+        self.assertFalse(self.resource.title in body)
```
Comment (carsongee) [inline search/tests/test_search_view.py]: Please use the nice Django TestCase self.AssertNotContains: https://docs.djangoproject.com/en/1.8/topics/testing/tools/#django.test.SimpleTestCase.assertNotContains to also vet the status while you are at it.

**Comment by carsongee** [comment]
Other than the one thing, this looks good as long as coverage and tests come back ok.

**Comment by ShawnMilo** [comment]
Thanks, @carsongee. That fix actually showed that the test was bad (it was getting a 403 instead of a 200). Test fixed. Code still worked despite bad test, and had been tested manually as well.

**Comment by carsongee** [comment]
Just verified the regression, i.e. your test fails without your code changes, and all looks good.  Once tests pass :hospital:

**Pull request merged**